### PR TITLE
Remove reference to 2.x, 3.x, 4.x versions

### LINKED ISSUE #17583: Remove some references to 2.x, 3.x, 4.x version
Hi, 

I found some places were we have references to EOF versions.
Are you ok if we remove/rework these statements on 6.3 branch ?

## Symfony 2 references:

contributing/code/bc.rst

> This promise was introduced with Symfony 2.3 and does not apply to previous versions of Symfony.

## Symfony 3 references:

service_container.rst

> Prior to Symfony 3.3, all services and (typically) arguments were 

components/config/definition.rst

> ``ifEmpty()`` (since Symfony 3.2)

## Symfony 4 references:

profiler.rst

> The possibility to use a matcher to enable the profiler conditionally was removed in Symfony 4.0.

> Symfony Profiler cannot be enabled/disabled conditionally using matchers, because
that feature was removed in Symfony 4.0. However, you can use...

components/browser_kit.rst

> In Symfony versions prior to 4.3, the BrowserKit component could only make
    internal requests to your application. Starting from Symfony 4.3, this
    component can also...

components/event_dispatcher/container_aware_dispatcher.rst

> The ``ContainerAwareEventDispatcher`` was removed in Symfony 4.0. Use
    ``EventDispatcher`` with closure-proxy injection instead.

components/filesystem/lock_handler.rst

> The ``LockHandler`` utility was removed in Symfony 4.0. Use the new Symfony :doc:`Lock component </components/lock>` instead.

### PR DESCRIPTION
**Pull request by alamirault**
closes #17583

### CODE REVIEW

File: components/browser_kit.rst [hunk 1]
```diff
@@ -8,13 +8,6 @@ The BrowserKit Component
     The BrowserKit component simulates the behavior of a web browser, allowing
     you to make requests, click on links and submit forms programmatically.
 
-.. note::
```
Comment (alamirault) [inline components/browser_kit.rst:11]: Section doc already exists, we can drop note
https://symfony.com/doc/current/components/browser_kit.html#making-external-http-requests

File: components/filesystem/lock_handler.rst [hunk 2]
```diff
@@ -1,7 +0,0 @@
-LockHandler
```
Comment (alamirault) [inline components/filesystem/lock_handler.rst:1]: Found no reference to the removed files but I don't know side effects

File: service_container.rst [hunk 3]
```diff
@@ -1074,11 +1074,8 @@ unique string as the key of each service config:
 Explicitly Configuring Services and Arguments
 ---------------------------------------------
 
-Prior to Symfony 3.3, all services and (typically) arguments were explicitly configured:
-it was not possible to :ref:`load services automatically <service-container-services-load-example>`
-and :ref:`autowiring <services-autowire>` was much less common.
-
-Both of these features are optional. And even if you use them, there may be some
+:ref:`Load services automatically <service-container-services-load-example>`
+and :ref:`autowiring <services-autowire>` are optional. And even if you use them, there may be some
```
Comment (alamirault) [inline service_container.rst]: IMO we can do better than my rewording

File: profiler.rst [hunk 4]
```diff
@@ -110,16 +110,10 @@ need to create a custom data collector. Instead, use the built-in utilities to
     Consider using a professional profiler such as `Blackfire`_ to measure and
     analyze the execution of your application in detail.
 
-Enabling the Profiler Conditionally
+Enabling the Profiler programmatically
 -----------------------------------
```
Comment (OskarStark) [inline profiler.rst]: * [ ] programatically` -> `Programatically` (uppercase P)
* [ ] Underline length should math the headline length

File: service_container.rst [hunk 5]
```diff
@@ -1074,11 +1074,8 @@ unique string as the key of each service config:
 Explicitly Configuring Services and Arguments
 ---------------------------------------------
 
-Prior to Symfony 3.3, all services and (typically) arguments were explicitly configured:
-it was not possible to :ref:`load services automatically <service-container-services-load-example>`
-and :ref:`autowiring <services-autowire>` was much less common.
-
-Both of these features are optional. And even if you use them, there may be some
+:ref:`Load services automatically <service-container-services-load-example>`
```
Comment (OskarStark) [inline service_container.rst]:
```suggestion
:ref:`Loading services automatically <service-container-services-load-example>`
```
?

File: profiler.rst [hunk 6]
```diff
@@ -110,16 +110,10 @@ need to create a custom data collector. Instead, use the built-in utilities to
     Consider using a professional profiler such as `Blackfire`_ to measure and
     analyze the execution of your application in detail.
 
-Enabling the Profiler Conditionally
```
Comment (OskarStark) [inline profiler.rst:113]: * [ ] Please add the following above the headline, to make the old link work again:

`.. _enabling-the-profiler-conditionally:`

**Review by OskarStark — APPROVED** [review]
@xabbuh not sure about the removal of the 2 files

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
looks good to me, I don't think we need redirects for the two removed chapters

@javiereguiluz @wouterj What do you think?

**Comment by javiereguiluz** [comment]
I agree with @xabbuh: we "soft removed" those two articles in 4.0 version, so we can now fully remove them.

I'm merging this in 6.0 branch and up. Thanks Antoine for an excellent contribution!

**Pull request merged**